koku/masu/processor/tasks.py

```diff
@@ -349,6 +350,7 @@ def update_summary_tables(  # noqa: C901
     synchronous=False,
     tracing_id=None,
     ocp_on_cloud=True,
+    manifest_list=None,
 ):
```

Comment:
So this function now takes both a `manifest_id` and a `manifest_list`? I'm not sure I understand why both should be inputs to the function, shouldn't it just accept one or the other?

Reply:
Sounds good I will get started on the changes. 🙇

Reply:
I'm not sure I know what changes you are talking about, are you going to change the function to only accept one of those two parameters?

Reply:
So, I was exploring the idea of passing a `manifest_list` into the `ReportSummaryUpdater` class. However, that started to feel like a can of worms. 

### Why we pass in manifest_id as an optional parameter

I did discover why the manifest_id is passed in to that class though, we essentially use the manifest in some of our cloud specific (AWSReportParquetSummaryUpdater) classes in the `_get_sql_inputs` method to decide the start & end date for our summary sql. We don't actually use it for GCP & OCI, but since it is used in some of the child classes we have to pass it to all of the child classes. 

### Why I needed to pass in manifest list

So, GCP is the only downloader that returns a list of manifests instead of one. Part of the work I am doing in this pull request is no longer call a summary task per manifest by moving [this section](https://github.com/project-koku/koku/blob/65cb6dfb51dcb179fcfda3a2698c1fad7c25b9a3/koku/masu/processor/orchestrator.py#L243-L254) outside of the for loop for `manifest_list` which creates our processing task.

However, now that we have one summary task for GCP, I ran into an issue where only one manifest is being marked as complete since we mark manifests as complete at the end of the summary task. My solution to this was to pass in a manifest_list variable and update method that marks them to accept a list. 

Thanks for the question @cgoodfred

I think since they are both optional parameters I am gonna leave as is for now.